sqlmesh/lsp/main.py

```diff
@@ -97,7 +177,15 @@ def ensure_context_for_document(self, document: TextDocument) -> TextDocument:
                 if config_path.exists():
                     try:
                         # Use user-provided instantiator to build the context
-                        self.context = self.context_class(paths=[path])
+                        context = self.context_class(paths=[path])
+                        self.context_and_reverse_map = (
+                            context,
+                            {
+                                f"file://{Path(model._path).resolve().as_posix()}": model.name
+                                for model in context._models.values()
```

Comment:
This looks like an exact same thing as 158-164. I think there should be only one way to populate the map.